core/src/main/java/org/apache/carbondata/core/carbon/metadata/schema/table/CarbonTable.java

```diff
@@ -504,4 +521,19 @@ public void setBlockSizeInMB(int blockSize) {
     this.blockSize = blockSize;
   }
 
+  /**
+   * to get the all dimension of a table
+   *
+   * @param tableName
+   * @return all dimension of a table
+   */
+  public CarbonDimension getAllDimensionByName(String tableName, String columnName) {
```

Comment:
it is used to get normal dimension and the primitive dimenson of the complex. 
I will rename it to a proper name